homeassistant/helpers/entity.py

```diff
@@ -884,6 +894,44 @@ def _async_write_ha_state(self) -> None:
             hass.states.async_set(
                 entity_id, STATE_UNKNOWN, {}, self.force_update, self._context
             )
+        if not state_updated:
+            return
+
+        if entry:
+            # Make sure capabilities in the entity registry are up to date. Capabilities
+            # include capability attributes, device class and supported features
+            original_device_class: str | None = self.device_class
```

Comment:
```suggestion
            original_device_class: str | None = attr.get(ATTR_DEVICE_CLASS)
```

We can avoid recalculating it again

Reply:
May have to save it in a local var before customize though unless we want customize to apply

Reply:
I guess customize should apply though so probably good as suggested

Reply:
Not like that, as discussed on discord, but we can let `_async_generate_attributes` return a dict with shadowed properties

Reply:
Commit [b136b54](https://github.com/home-assistant/core/pull/101748/commits/b136b541bb15fd2c3ce5ea68f56706020f186cab) stores the device class in a dict with shadowed properties